lib/matplotlib/axes/_base.py

```diff
@@ -146,12 +147,17 @@ def __getstate__(self):
 
     def __setstate__(self, state):
         self.__dict__ = state.copy()
-        self.set_color_cycle()
-
-    def set_color_cycle(self, clist=None):
-        if clist is None:
-            clist = rcParams['axes.color_cycle']
-        self.color_cycle = itertools.cycle(clist)
+        self.set_prop_cycle()
+
+    def set_prop_cycle(self, prop_cycler=None):
+        if prop_cycler is None:
+            prop_cycler = rcParams['axes.prop_cycle']
+            if prop_cycler is None and 'axes.color_cycle' in rcParams:
+                clist = rcParams['axes.color_cycle']
+                prop_cycler = cycler('color', clist)
+        self.prop_cycler = itertools.cycle(prop_cycler)
+        # Make a copy
+        self._prop_keys = list(prop_cycler.keys)
```

Comment:
Sorry, I am being a bit defensive.  

I spent a while thinking about what keys should look like and chose set because there is no order associated with the keys anywhere. Turning keys into a list is undoing something that I intentionally did :stuck_out_tongue_winking_eye: .  

The set coming out of the call to `c.keys` is a copy of the internal set, I think for almost exactly this reason.  I was more worried about the outside world getting a ref to the internal state and popping keys off or something like that.